docs/framework/20categories/30implementation/verified-code.md

```diff
@@ -1,3 +1,19 @@
 ### Verified Code on Block Explorer
 
-The source code for deployed contracts should be verified and made available in the network's primary block explorer (e.g., EtherScan for Ethereum, SnowTrace for Avalanche). Not being able to determine what code has been deployed reduces the trust stakeholders have in the project. 
\ No newline at end of file
+Source code verification for all deployed smart contracts is critical for user safety. It acts as an assurance of correctness of a smart contract by ensuring that a protocol’s submitted source code is the same as the one executed at the contract address. Moreover, it allows users interacting with a protocol’s smart contract to read and audit it. This enables them to determine what code has been deployed and what types of risks the user is taking by interacting with the protocol. As a result, if a protocol’s code isn’t verified, it is considered less transparent and should be trusted less by users and developers. 
+
+*Considerations:*
+* What tool has been used to verify the source code?
+
+Source code can be verified by different tools such as Etherscan, Sourcify, Tenderly, etc., but not all tools are created equal. The level of confidence around the source code verification must be assessed based on the tool used. For instance, for contracts deployed on Ethereum, [Etherscan’s verify contract code tool](https://etherscan.io/verifyContract) is considered a trustworthy tool.
+* Which module of the source code is verified? Does it cover all the key on-chain components?
+
+It’s critical for the source code of all key smart contracts of a protocol to be verified. It’s possible for protocols to verify only certain smart contracts while hiding malicious code to give a false perception of its smart contracts being verified.
+* Could the protocol be less transparent about their source code on purpose? 
+
+The rule should always be to trust, but verify. If the source code isn’t verified, a protocol can be a potential rug, hiding the real intention of their smart contract. Or, the protocol could be following the practice of ‘security through obscurity', keeping the code unverified and thus private to prevent people from learning about how a specific feature works and potentially misusing it.
```

Comment:
nit: it's highly unlikely the audience won't know what a "rug" is, but adding a link to a definition might be useful.